walrus_mainboard_driver/src/walrus_mainboard_driver/walrus_mainboard_driver.cpp

```diff
@@ -0,0 +1,670 @@
+		aei.registerHandle(effort_handle1);
+		hardware_interface::ActuatorHandle effort_handle2(state_handle2, &pod_effort_cmd[BR-1]);
+		aei.registerHandle(effort_handle2);
+		hardware_interface::ActuatorHandle effort_handle3(state_handle3, &pod_effort_cmd[BL-1]);
+		aei.registerHandle(effort_handle3);
+		
+		//Add diagnostics updaters
+		diagnostic_updater.setHardwareID("Walrus Main Board");
+		diagnostic_updater.add("Chassis", this, &MainBoardDriver::chassis_diagnostic_callback);
+		diagnostic_updater.add("Front Left Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, FL-1));
+		diagnostic_updater.add("Front Right Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, FR-1));
+		diagnostic_updater.add("Back Right Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, BR-1));
+		diagnostic_updater.add("Back Left Pod", boost::bind(&MainBoardDriver::pod_diagnostic_callback, this, _1, BL-1));
+		diagnostic_updater.add("Main Control Board", this, &MainBoardDriver::mainboard_diagnostic_callback);
+		diagnostic_updater.add("Battery 1", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 0));
+		diagnostic_updater.add("Battery 2", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 1));
+		diagnostic_updater.add("Battery 3", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 2));
+		diagnostic_updater.add("Battery 4", boost::bind(&MainBoardDriver::batt_diagnostic_callback, this, _1, 3));
+		diagnostic_updater.add("Power Systems", this, &MainBoardDriver::power_diagnostic_callback);
+		diagnostic_updater.add("Left Drive Motor Temperature", this, &MainBoardDriver::left_drive_temp_diagnostic_callback);
+		diagnostic_updater.add("Right Drive Motor Temperature", this, &MainBoardDriver::right_drive_temp_diagnsotic_callback);
+	}
+	
+	bool MainBoardDriver::init()
+	{	    
+	    //Setup publishers and subscribers to communicate with the embedded board
+		hs_control = nh.advertise<walrus_firmware_msgs::MainBoardHighSpeedControl>("/walrus/main_board/hs_control", 1000);
+		to_board = nh.advertise<walrus_firmware_msgs::MainBoardControl>("/walrus/main_board/PC_to_board_control", 1000);
+		hs_feedback = nh.subscribe("/walrus/main_board/hs_feedback", 1000, &MainBoardDriver::hs_feedback_callback, this);
+		ls_data = nh.subscribe("/walrus/main_board/ls_data", 1000, &MainBoardDriver::ls_data_callback, this);
+		from_board = nh.subscribe("/walrus/main_board/board_to_PC_control", 1000, &MainBoardDriver::from_board_callback, this);
+	    return true;
+	}
+	
+	void MainBoardDriver::read()
+	{
+        double temp, dr;
+	    double interval = (ros::Time::now() - last_read).toSec();
+	    last_read = ros::Time::now();
+	    
+	    {
+	        boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	       
+	        for (int l = 0; l < 4; l++)
+	        {
+		        temp = (hs_feedback_msg.pod_position[l]/1024.0)*2*PI;
```

Comment:
Can you name this something meaningful. Also if the values of temp and dr have no meaning outside of the for loop and don't need to persist between iterations then they should be declared inside the loop.